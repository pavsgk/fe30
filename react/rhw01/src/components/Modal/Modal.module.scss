.ModalBackground {
  background-color: rgba(0, 0, 0, 0.3);
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  display: flex;
  justify-content: center;
  align-items: center;
}

.Modal {
  font-family: 'Open-sans', sans-serif;
  background-color: #E74C3C;
  border-radius: 5px;
  max-width: 515px;

  button {
    cursor: pointer;
  }
}

.ModalContent {
  min-height: 185px;
  display: flex;
  padding: 0 30px;
  flex-direction: column;
  text-align: center;
  justify-content: space-around;
  
  p {
    font-size: 15px;
    line-height: 30px;
    color: #FFF;
  }
}

.ModalHeader {
  min-height: 70px;
  padding: 0 30px;
  background-color: rgba(0, 0, 0, .15);
  display: flex;
  justify-content: space-between;
  align-items: center;
  border-top-left-radius: 5px;
  border-top-right-radius: 5px;

  h2 {
    font-weight: bold;
    font-size: 22px;
    line-height: 60px;
    color: #FFF;
    -webkit-text-stroke: 1px black;
  }

  button {
    font-size: 30px;
    border: 0;
    background-color: transparent;
  }
}
