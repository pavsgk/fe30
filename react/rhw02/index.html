<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Document</title>
</head>
<body>

<script>
  const sendReq = async () => {
    const result = await fetch('https://script.google.com/macros/s/AKfycbzEYFV5JBtD5d2D2swAs6i9n0SeqMxAgZRD-tJG6F3ONbSgF38kL4qeuJ-U-FMDebvxew/exec?___testS1hop1').then(res => res.json());
    console.log(result);
    return result;
  }

  sendReq();



</script>  
</body>
</html>