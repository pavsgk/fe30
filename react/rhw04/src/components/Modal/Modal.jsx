import React from "react";
import styles from './Modal.module.scss';
import { useDispatch, useSelector } from "react-redux";
import { hideModal } from "../../store/actionCreators";


function Modal() {
  
  const dispatch = useDispatch();

  const modal = useSelector(state => state.modal);

  if (!modal.isShown) return null;

  const { actionFn, title, text } = modal;
  const hasCloseButton = true;
  const optionalCeneterd = {
    'justifyContent': hasCloseButton ? 'space-between' : 'center',
  }


  return (
    <div className={styles.ModalBackground} onClick={() => dispatch(hideModal())}>
      <div className={styles.Modal} onClick={e => e.stopPropagation()}>
        <header className={styles.ModalHeader} style={optionalCeneterd}>
          <h2>{title}</h2>
          {hasCloseButton && <button onClick={() => dispatch(hideModal())}>&#10761;</button>}
        </header>
        <div className={styles.ModalContent}>
          <p>{text}</p>
          <div className={styles.ModalControls}>
            <button onClick={() => {
              actionFn();
              dispatch(hideModal())
            }}>Yes</button>
            <button onClick={() => dispatch(hideModal())}>No</button>
          </div>
        </div>
      </div>
    </div>
  )
}

export default Modal;