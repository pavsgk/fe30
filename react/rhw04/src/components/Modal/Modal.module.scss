.ModalBackground {
  background-color: rgba(0, 0, 0, 0.3);
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  display: flex;
  justify-content: center;
  align-items: center;
}

.Modal {
  font-family: 'Open-sans', sans-serif;
  background-color: lightgray;
  border-radius: 10px;
  max-width: 515px;

  button {
    cursor: pointer;
  }
}

.ModalContent {
  min-height: 185px;
  display: flex;
  padding: 0 30px;
  flex-direction: column;
  text-align: center;
  justify-content: space-evenly;
  gap: 10px;
  
  p {
    font-size: 20px;
    line-height: 30px;
    color: #000;
  }
  
  .ModalControls {
    display: flex;
    justify-content: space-around;

    button {
      padding: 10px;
      min-width: 60px;
      border-radius: 10px;
      font-size: 20px;

      &:hover {
        filter: drop-shadow(0 0 10px #4444dd)
      }
    }
  }
}

.ModalHeader {
  min-height: 70px;
  padding: 0 30px;
  background-color: rgba(0, 0, 0, .15);
  display: flex;
  justify-content: space-between;
  align-items: center;
  border-top-left-radius: 5px;
  border-top-right-radius: 5px;

  h2 {
    font-weight: bold;
    font-size: 22px;
    line-height: 60px;
    color: #FFF;
    -webkit-text-stroke: 1px black;
  }

  button {
    font-size: 30px;
    border: 0;
    background-color: transparent;
  }
}