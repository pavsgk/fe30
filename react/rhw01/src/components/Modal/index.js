import { PureComponent } from "react";
import App from "../../App";
import styles from './Modal.module.scss';


class Modal extends PureComponent {
  render () {
    const { title, text, hideFn, id } = this.props;
    return (
      <div className={styles.ModalBackground} onClick={e => {
        if (e.currentTarget === e.target) hideFn(e, id);
      }}>
        <div className={styles.Modal}>
          <header className={styles.ModalHeader}><h2>{title}</h2><button onClick={e => hideFn(e, id)}>&#10761;</button></header>
          <div className={styles.ModalContent}>
            <p>{text}</p>
            <div>
              <button>OK</button>
              <button>Cancel</button>
            </div>
          </div>
        </div>
      </div>
    )
  }
}

export default Modal;