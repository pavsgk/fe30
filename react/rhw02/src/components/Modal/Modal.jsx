import { PureComponent } from "react";
import styles from './Modal.module.scss';
import PropTypes from 'prop-types';


class Modal extends PureComponent {

  render () {
    const { actionFn, title, text, hasCloseButton, hideFn } = this.props;
    const optionalCeneterd = {
      'justifyContent': hasCloseButton ? 'space-between' : 'center',
    }
    return (
      <div className={styles.ModalBackground} onClick={() => hideFn()}>
        <div className={styles.Modal} onClick={e => e.stopPropagation()}>
          <header className={styles.ModalHeader} style={optionalCeneterd}>
            <h2>{title}</h2>
            {hasCloseButton && <button onClick={() => hideFn()}>&#10761;</button>}
          </header>
          <div className={styles.ModalContent}>
            <p>{text}</p>
            <div>
              <button onClick={() => {
                actionFn();
                hideFn();
              }}>Yes</button>
              <button onClick={() => hideFn()}>No</button>
            </div>
          </div>
        </div>
      </div>
    )
  }
}

Modal.propTypes = {
  actionFn: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  hideFn: PropTypes.func.isRequired,
  text: PropTypes.string.isRequired,
  hasCloseButton: PropTypes.bool,
}

Modal.defaultProps = {
  hasCloseButton: true
}

export default Modal;