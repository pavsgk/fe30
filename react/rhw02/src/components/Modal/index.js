import { PureComponent } from "react";
import styles from './Modal.module.scss';


class Modal extends PureComponent {

  render () {
    const { actionFn, title, text, hasCloseButton, hideFn } = this.props;
    const optionalCeneterd = {
      'justifyContent': hasCloseButton ? 'space-between' : 'center',
    }
    return (
      <div className={styles.ModalBackground} onClick={() => hideFn()}>
        <div className={styles.Modal} onClick={e => e.stopPropagation()}>
          <header className={styles.ModalHeader} style={optionalCeneterd}>
            <h2>{title}</h2>
            {hasCloseButton && <button onClick={() => hideFn()}>&#10761;</button>}
          </header>
          <div className={styles.ModalContent}>
            <p>{text}</p>
            <div>
              
            </div>
          </div>
        </div>
      </div>
    )
  }
}

export default Modal;