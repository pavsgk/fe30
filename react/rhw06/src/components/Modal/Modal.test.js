import '@testing-library/jest-dom';
import Modal from "./Modal";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";

const testStore = {
  modal: {
    isShown: true,
    actionFn: () => {},
    text: "This is a test modal",
    title: "Test modal",
  },
  getState() {
    return this;
  },
  subscribe() {},
  dispatch() {},
};

describe("Snapshot test", () => {
  test("Should component render", () => {

    const { asFragment } = render(
      <Provider store={testStore}>
        <Modal />
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test("Should component hide", () => {

    testStore.modal.isShown = false;

    const { queryByText } = render(
      <Provider store={testStore}>
        <Modal />
      </Provider>)

    expect(queryByText(testStore.modal.title)).not.toBeInTheDocument();
  
  });

});