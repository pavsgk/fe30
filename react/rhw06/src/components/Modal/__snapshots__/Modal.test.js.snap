// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test Should component render 1`] = `
<DocumentFragment>
  <div
    class="ModalBackground"
  >
    <div
      class="Modal"
    >
      <header
        class="ModalHeader"
        style="justify-content: space-between;"
      >
        <h2>
          Test modal
        </h2>
        <button>
          â¨‰
        </button>
      </header>
      <div
        class="ModalContent"
      >
        <p>
          This is a test modal
        </p>
        <div
          class="ModalControls"
        >
          <button>
            Yes
          </button>
          <button>
            No
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
